### PR TITLE
New UI

### PR DESCRIPTION
**Pull request by asafgabai**
- [ ] All [tests](https://github.com/jfrog/jfrog-idea-plugin/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
-----
Depends on:
https://github.com/jfrog/build-info/pull/688
https://github.com/jfrog/ide-plugins-common/pull/95

### CODE REVIEW

File: src/main/java/com/jfrog/ide/idea/ui/JFrogCiToolWindow.java [hunk 1]
```diff
@@ -49,33 +57,72 @@ public class JFrogCiToolWindow extends AbstractJFrogToolWindow {
...
 
-    @Override
-    String getComponentsTreeTitle() {
+    private String getComponentsTreeTitle() {
         return " Build Components (Issues #)";
     }
 
-    @Override
+    /**
+     * Create CI or local issues filter menu
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/JFrogCiToolWindow.java:92]: Same comment for the other methods in this file
```suggestion
     * Create CI issues filter menu
```

File: src/main/java/com/jfrog/ide/idea/ui/JFrogLocalToolWindow.java [hunk 2]
```diff
@@ -1,104 +1,175 @@
...
-import com.jfrog.ide.idea.ui.menus.filtermenu.*;
+import com.jfrog.ide.idea.log.Logger;
+import com.jfrog.ide.idea.ui.jcef.message.MessagePacker;
 import com.jfrog.ide.idea.ui.utils.ComponentUtils;
+import com.jfrog.ide.idea.ui.webview.WebviewObjectConverter;
+import org.apache.commons.io.FileUtils;
+import org.cef.browser.CefBrowser;
+import org.cef.browser.CefFrame;
+import org.cef.handler.CefLoadHandlerAdapter;
+import com.jfrog.ide.idea.ui.webview.model.ExtendedInformation;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/JFrogLocalToolWindow.java:26]: Please reorganize imports

File: src/main/java/com/jfrog/ide/idea/ui/JFrogLocalToolWindow.java [hunk 3]
```diff
@@ -1,104 +1,175 @@
...
-        JLabel title = new JBLabel(" More Info");
-        title.setFont(title.getFont().deriveFont(TITLE_FONT_SIZE));
-
-        moreInfoPanel = new JBPanel<>(new BorderLayout()).withBackground(UIUtil.getTableBackground());
-        String panelText = supported ? ComponentUtils.SELECT_COMPONENT_TEXT : ComponentUtils.UNSUPPORTED_TEXT;
-        moreInfoPanel.add(ComponentUtils.createDisabledTextLabel(panelText), BorderLayout.CENTER);
-        issuesDetailsScroll = ScrollPaneFactory.createScrollPane(moreInfoPanel, ScrollPaneConstants.VERTICAL_SCROLLBAR_ALWAYS, ScrollPaneConstants.HORIZONTAL_SCROLLBAR_NEVER);
-        return new TitledPane(JSplitPane.VERTICAL_SPLIT, TITLE_LABEL_SIZE, title, issuesDetailsScroll);
+
+        Path tempDirPath;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/JFrogLocalToolWindow.java:108]: Let's delete this directory in `dispose()`

File: src/main/java/com/jfrog/ide/idea/ui/JFrogLocalToolWindow.java [hunk 4]
```diff
@@ -1,104 +1,175 @@
...
+        JScrollPane treeScrollPane = ScrollPaneFactory.createScrollPane(treePanel);
+        treeScrollPane.getVerticalScrollBar().setUnitIncrement(SCROLL_BAR_SCROLLING_UNITS);
+        treeScrollPane.setBorder(null);
+        return treeScrollPane;
+    }
+
+    private void updateIssueOrLicenseInWebview(VulnerabilityOrViolation vulnerabilityOrViolation) {
+        if (vulnerabilityOrViolation instanceof Issue) {
+            Issue issue = (Issue) vulnerabilityOrViolation;
+            messagePacker.send("DEPENDENCY", WebviewObjectConverter.convertIssueToDepPage(issue));
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/JFrogLocalToolWindow.java:154]: The DEPENDENCY should be in a final static String class variable

File: src/main/java/com/jfrog/ide/idea/ui/jcef/message/MessagePacker.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    public MessagePacker(JBCefBrowserBase browser) {
+        this.browser = browser;
+    }
+
+    public void send(String type, Object data) {
+        try {
+            String raw = MessagePipeSupport.Pack(type, data);
+            this.send(raw);
+        } catch (JsonProcessingException e) {
+            System.out.println(e.getMessage());
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/jcef/message/MessagePacker.java:20]: Please use the logger

File: src/main/java/com/jfrog/ide/idea/ui/jcef/message/MessagePacker.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        }
+    }
+
+    private void send(String raw) {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/jcef/message/MessagePacker.java:24]:
```suggestion
    private void executeJavaScript(String raw) {
```

File: src/main/java/com/jfrog/ide/idea/ui/jcef/message/MessagePipe.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+/**
+ * Defines base interface for message channel implementation.
+ * Both IDE and browser implementations must be in sync with each other.
+ */
+public interface MessagePipe {
+    /**
+     * Sends raw message with type [type] and content [data] to the other end.
+     */
+    public void send(String type, Object data);
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/jcef/message/MessagePipe.java:11]: Unnecessary in interfaces
```suggestion
    void send(String type, Object data);
```

File: src/main/java/com/jfrog/ide/idea/ui/menus/filtermenu/IssueFilterMenu.java [hunk 8]
```diff
@@ -12,8 +12,7 @@
 /**
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/menus/filtermenu/IssueFilterMenu.java:12]: I've found some unused imports in this file

File: src/main/java/com/jfrog/ide/idea/ui/utils/ComponentUtils.java [hunk 9]
```diff
@@ -8,7 +8,6 @@
 import com.intellij.util.ui.JBInsets;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/utils/ComponentUtils.java:8]: I've found some unused imports in this file

File: src/main/java/com/jfrog/ide/idea/ui/webview/WebviewObjectConverter.java [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+import com.jfrog.ide.idea.ui.webview.model.*;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class WebviewObjectConverter {
+    public static DependencyPage convertIssueToDepPage(Issue issue) {
+        ExtendedInformation extendedInformation = null;
+        if (issue.getResearchInfo() != null) {
+            com.jfrog.ide.common.tree.ResearchInfo issueResearchInfo = issue.getResearchInfo();
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/WebviewObjectConverter.java:16]:
```suggestion
            ResearchInfo issueResearchInfo = issue.getResearchInfo();
```

File: src/main/java/com/jfrog/ide/idea/ui/webview/WebviewObjectConverter.java [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+                convertVersionRanges(issue.getFixedVersions()),
+                convertVersionRanges(issue.getInfectedVersions()),
+                convertReferences(issue.getReferences()),
+                new Cve(
+                        issue.getCve().getCveId(),
+                        issue.getCve().getCvssV2Score(),
+                        issue.getCve().getCvssV2Vector(),
+                        issue.getCve().getCvssV3Score(),
+                        issue.getCve().getCvssV3Vector()
+                ),
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/WebviewObjectConverter.java:51]: Let's create a translator from com.jfrog.ide.common.tree.Cve com.jfrog.ide.idea.ui.webview.model.Cve

File: src/main/java/com/jfrog/ide/idea/ui/webview/WebviewObjectConverter.java [hunk 12]
```diff
@@ -0,0 +1,213 @@
...
+import com.jfrog.ide.common.tree.Artifact;
+import com.jfrog.ide.common.tree.ImpactTreeNode;
+import com.jfrog.ide.common.tree.Issue;
+import com.jfrog.ide.common.tree.LicenseViolation;
+import com.jfrog.ide.idea.ui.webview.model.*;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class WebviewObjectConverter {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/WebviewObjectConverter.java:12]: The methods in this file do not comply with our standards.
I suggest doing the following:
1.  Splitting long methods into smaller methods
2. Using "for in" loops over old style "for int" loops.
3. Use apache-commons libraries when possible

File: src/main/java/com/jfrog/ide/idea/ui/webview/WebviewObjectConverter.java [hunk 13]
```diff
@@ -0,0 +1,213 @@
...
+        return new ImpactedPath(removeComponentIdPrefix(impactTreeNode.getName()), children);
+    }
+
+    private static String removeComponentIdPrefix(String compId) {
+        final String prefixSeparator = "://";
+        int prefixIndex = compId.indexOf(prefixSeparator);
+        if (prefixIndex == -1) {
+            return compId;
+        }
+        return compId.substring(prefixIndex + prefixSeparator.length());
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/WebviewObjectConverter.java:98]:
```suggestion
        return StringUtils.substringAfter(compId, "://");
```

File: src/main/java/com/jfrog/ide/idea/ui/webview/WebviewObjectConverter.java [hunk 14]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+    private static String[] convertVersionRanges(List<String> xrayVerRanges) {
+        if (xrayVerRanges == null) {
+            return new String[0];
+        }
+        return xrayVerRanges.stream().map(s -> convertVersionRange(s)).toArray(String[]::new);
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/WebviewObjectConverter.java:105]:
```suggestion
        return xrayVerRanges.stream().map(WebviewObjectConverter::convertVersionRange).toArray(String[]::new);
```

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/ApplicableDetails.java [hunk 15]
```diff
@@ -0,0 +1,19 @@
+package com.jfrog.ide.idea.ui.webview.model;
+
+import java.io.Serializable;
+
+public class ApplicableDetails implements Serializable {
+    boolean isApplicable;
+    String applicableFixReason;
+    String filePathEvidence;
+    String codeEvidence;
+    String searchTarget;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/ApplicableDetails.java:10]: All of them should be private

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/Cve.java [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+
+import java.io.Serializable;
+
+public class Cve implements Serializable {
+    String id;
+    String cvssV2Score;
+    String cvssV2Vector;
+    String cvssV3Score;
+    String cvssV3Vector;
+    ApplicableDetails applicableData;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/Cve.java:11]: Should be private

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/Cve.java [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/Cve.java:31]:
```suggestion
```

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/Cve.java [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+
+    public String getCvssV2Vector() {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/Cve.java:33]: Let's add this in all places whereby we have an unused methods with a justified reason
```suggestion
    @SuppressWarnings("unused")
    public String getCvssV2Vector() {
```

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/DependencyPage.java [hunk 19]
```diff
@@ -0,0 +1,169 @@
...
+    License[] license;
+    String summary;
+    String[] fixedVersion;
+    String[] infectedVersion;
+    Reference[] references;
+    Cve cve;
+    ImpactedPath impactedPath;
+    String[] watchName;
+    String edited;
+    ExtendedInformation extendedInformation;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/DependencyPage.java:20]: Should be private

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/DependencyPage.java [hunk 20]
```diff
@@ -0,0 +1,169 @@
+package com.jfrog.ide.idea.ui.webview.model;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/DependencyPage.java:1]: I think this file deserves a builder.

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/DependencyPage.java [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+        this.watchName = watchName;
+        this.edited = edited;
+        this.extendedInformation = extendedInformation;
+    }
+
+    public License[] getLicense() {
+        return license;
+    }
+
+    public void setLicense(License[] license) {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/DependencyPage.java:50]: Many compiler warnings about unused methods

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/DependencyPage.java [hunk 22]
```diff
@@ -0,0 +1,169 @@
...
+
+    public ExtendedInformation getExtendedInformation() {
+        return extendedInformation;
+    }
+
+    public void setExtendedInformation(ExtendedInformation extendedInformation) {
+        this.extendedInformation = extendedInformation;
+    }
+
+
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/DependencyPage.java:94]:
```suggestion
```

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/DependencyPage.java [hunk 23]
```diff
@@ -0,0 +1,169 @@
...
+    public Cve getCve() {
+        return cve;
+    }
+
+    public void setCve(Cve cve) {
+        this.cve = cve;
+    }
+
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/DependencyPage.java:103]:
```suggestion
```

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/DependencyPage.java [hunk 24]
```diff
@@ -0,0 +1,169 @@
+package com.jfrog.ide.idea.ui.webview.model;
+
+import java.io.Serializable;
+
+public class DependencyPage implements Serializable {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/DependencyPage.java:5]:
```suggestion
public class DependencyPage {
```

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/Cve.java [hunk 25]
```diff
@@ -0,0 +1,64 @@
+package com.jfrog.ide.idea.ui.webview.model;
+
+import java.io.Serializable;
+
+public class Cve implements Serializable {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/Cve.java:5]: No serialization happen
```suggestion
public class Cve {
```

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/ApplicableDetails.java [hunk 26]
```diff
@@ -0,0 +1,19 @@
+package com.jfrog.ide.idea.ui.webview.model;
+
+import java.io.Serializable;
+
+public class ApplicableDetails implements Serializable {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/ApplicableDetails.java:5]:
```suggestion
public class ApplicableDetails {
```

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/ExtendedInformation.java [hunk 27]
```diff
@@ -0,0 +1,63 @@
+package com.jfrog.ide.idea.ui.webview.model;
+
+import java.io.Serializable;
+
+public class ExtendedInformation implements Serializable {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/ExtendedInformation.java:5]:
```suggestion
public class ExtendedInformation {
```

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/ExtendedInformation.java [hunk 28]
```diff
@@ -0,0 +1,63 @@
+package com.jfrog.ide.idea.ui.webview.model;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/ExtendedInformation.java:1]: This file is not according to our standards too

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/ImpactedPath.java [hunk 29]
```diff
@@ -0,0 +1,32 @@
+package com.jfrog.ide.idea.ui.webview.model;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/ImpactedPath.java:1]: This file is not according to our standards

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/JfrogResearchSeverityReason.java [hunk 30]
```diff
@@ -0,0 +1,42 @@
+package com.jfrog.ide.idea.ui.webview.model;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/JfrogResearchSeverityReason.java:1]: This file is not according to our standards

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/License.java [hunk 31]
```diff
@@ -0,0 +1,24 @@
+package com.jfrog.ide.idea.ui.webview.model;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/License.java:1]: This file is not according to our standards

File: src/main/java/com/jfrog/ide/idea/ui/webview/model/Reference.java [hunk 32]
```diff
@@ -0,0 +1,32 @@
+package com.jfrog.ide.idea.ui.webview.model;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/webview/model/Reference.java:1]: This file is not according to our standards

File: src/main/resources/META-INF/plugin.xml [hunk 33]
```diff
@@ -42,15 +42,14 @@
...
-        <projectService serviceImplementation="com.jfrog.ide.idea.ui.menus.filtermanager.LocalFilterManager"/>
         <projectService serviceImplementation="com.jfrog.ide.idea.ui.menus.filtermanager.CiFilterManager"/>
         <projectService serviceImplementation="com.jfrog.ide.idea.scan.ScanManagersFactory"/>
         <projectService serviceImplementation="com.jfrog.ide.idea.ci.CiManager"/>
         <projectService serviceImplementation="com.jfrog.ide.idea.ui.LocalComponentsTree"/>
         <projectService serviceImplementation="com.jfrog.ide.idea.ui.CiComponentsTree"/>
         <projectService serviceImplementation="com.jfrog.ide.idea.ui.JFrogToolWindow"/>
         <projectService serviceImplementation="com.jfrog.ide.idea.navigation.NavigationService"/>
-        <toolWindow id="JFrog" anchor="bottom" icon="/icons/jfrog_icon.png"
+        <toolWindow id="JFrog" anchor="right" icon="/icons/jfrog_icon.png"
```
Comment (yahavi) [inline src/main/resources/META-INF/plugin.xml:52]: I think we agreed to make in in the bottom

**Review by yahavi — CHANGES_REQUESTED** [review]
Well done @asafgabai!
Please make sure to not have any compiler warnings in the code. Ignoring the ones that look innocent, may hide some real issues in some cases.
Please **don't merge** before fixing compilation errors.

Also - I think we should refactor all files under ui/webview/model/

File: src/main/java/com/jfrog/ide/idea/scan/MavenScanManager.java [hunk 34]
```diff
@@ -165,23 +165,40 @@ private void updateChildrenNodes(DependencyTree parentNode, MavenArtifactNode ma
...
-     * Maven projects tree listener for scanning artifacts on dependencies changes.
-     */
-
-    private final class MavenProjectsTreeListener implements MavenProjectsTree.Listener {
-        @Override
-        public void projectResolved(@NotNull Pair<MavenProject, MavenProjectChanges> projectWithChanges,
-                                    NativeMavenProjectHolder nativeMavenProject) {
-            asyncScanAndUpdateResults();
+    @Override
+    protected List<FileTreeNode> groupArtifactsToDescriptorNodes(Collection<Artifact> depScanResults, Map<String, List<DependencyTree>> depMap) {
```
[the remark this answers is not in the document]
Comment (asafgabai) [reply src/main/java/com/jfrog/ide/idea/scan/MavenScanManager.java]: Currently, populateMavenModuleNode creates DependencyTree objects, not DescriptorFileTreeNodes.
I can move the creation of DescriptorFileTreeNodes to populateMavenModuleNode, but I think there are too many disadvantages to doing so:
1. In groupDependenciesToDescriptorNodes I create DescriptorFileTreeNodes only for descriptors that have issues. Moving the creation of them to populateMavenModuleNode will cause creating them for all descriptors (even if they don't have issues).
2. It'll also make us save treeNodeMap as a class variable, although we can avoid that.
3. The only thing we save by doing it, is the containsKey check, which is O(1).

File: src/main/java/com/jfrog/ide/idea/ui/JFrogLocalToolWindow.java (hunk 3 above)
Comment (asafgabai) [reply src/main/java/com/jfrog/ide/idea/ui/JFrogLocalToolWindow.java:107]: 3 lines below we call `deleteOnExit()`. Isn't it good enough?

File: src/main/java/com/jfrog/ide/idea/ui/jcef/message/MessagePacker.java (hunk 6 above)
Comment (asafgabai) [reply src/main/java/com/jfrog/ide/idea/ui/jcef/message/MessagePacker.java:24]: I actually think `send` is a better name for this method.

File: src/main/java/com/jfrog/ide/idea/scan/ScanManager.java [hunk 35]
```diff
@@ -119,37 +117,117 @@ protected void sendUsageReport() {
 
     protected abstract String getProjectPackageType();
 
+    protected abstract List<FileTreeNode> groupArtifactsToDescriptorNodes(Collection<Artifact> depScanResults, Map<String, List<DependencyTree>> depMap);
+
+    public abstract String getPackageType();
```
[the remark this answers is not in the document]
Comment (asafgabai) [reply src/main/java/com/jfrog/ide/idea/scan/ScanManager.java:131]: These methods don't necessarily return the same result. Currently, in YarnScanManager getProjectPackageType returns "yarn" and getPackageType returns "npm".
To make the difference clearer, I changed getProjectPackageType name to getPackageManagerName.

File: src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java [hunk 36]
```diff
@@ -161,16 +133,13 @@ public void refreshScanManagers(Utils.ScanLogicType scanLogicType, @Nullable Exe
...
+     * @param pkgType - package type name
+     * @param logger  - logger
      * @return Xray scan logic
      */
-    private ScanLogic createScanLogic(Utils.ScanLogicType type, ScanCache scanCache, Logger logger) {
-        if (type == Utils.ScanLogicType.GraphScan) {
-            return new GraphScanLogic(scanCache, logger);
-        }
-        return new ComponentSummaryScanLogic(scanCache, logger);
+    private ScanLogic createScanLogic(Utils.ScanLogicType type, String pkgType, Logger logger) {
```
[the remark this answers is not in the document]
Comment (asafgabai) [reply src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java:139]: Removing this parameter will cause getScanLogicType not to be called, so Xray version won't be checked. This thing needs a deeper examination. It'll be discussed in the coming days and fixed in another pull request.

**Comment by asafgabai** [comment]
[שינוי UI](https://trello.com/c/Qbw43Czs/61-%D7%A9%D7%99%D7%A0%D7%95%D7%99-ui)

**Pull request merged**